plugins/content/caption/base/lib/kWebVTTGenerator.php

```diff
@@ -49,40 +59,49 @@ public static function buildWebVTTM3U8File($segmentDuration, $entryDuration)
 			{
 				$result .= "#EXTINF:{$segmentDuration}.0,\r\n";
 			}
+
+			$result .= $linkParamsToConcat;
 			$result .= "segmentIndex/{$curIndex}.vtt\r\n";
 		}
 		$result .= "#EXT-X-ENDLIST\r\n";
 		return $result;
 	}
 
+	protected static function getManifestHeaderForWebVTT($segmentDuration)
+	{
+		$result = "#EXTM3U\r\n";
+		$result .= "#EXT-X-TARGETDURATION:{$segmentDuration}\r\n";
+		$result .= "#EXT-X-VERSION:3\r\n";
+		$result .= "#EXT-X-MEDIA-SEQUENCE:1\r\n";
+		$result .= "#EXT-X-PLAYLIST-TYPE:VOD\r\n";
+		return $result;
+	}
+
 	/**
 	 * @param array $parsedCaption
 	 * @param int $segmentIndex
 	 * @param int $segmentDuration
 	 * @param int $localTimestamp
 	 * @return string
 	 */
-	public static function buildWebVTTSegment(array $parsedCaption, $segmentIndex, $segmentDuration, $localTimestamp)
+	public static function buildWebVTTSegment(array $parsedCaption, $segmentIndex, $segmentDuration, $localTimestamp, $clipFrom, $clipTo)
 	{
-		$segmentStartTime = ($segmentIndex - 1) * $segmentDuration * 1000;
-		$segmentEndTime = $segmentIndex * $segmentDuration * 1000;
+		$result = self::getSegmentHeaderForWebVTT($localTimestamp);
 
-		$result = "WEBVTT\n";
-		if ($localTimestamp != 10000)
-		{
-			$result .= "X-TIMESTAMP-MAP=MPEGTS:900000,LOCAL:" . self::formatWebVTTTimeStamp($localTimestamp) . "\n";
-		}
-		$result .= "\n";
+		[$segmentStartTime, $segmentEndTime] = self::calculateSegmentTime($segmentIndex, $segmentDuration, $clipFrom, $clipTo);
 
 		foreach ($parsedCaption as $curCaption)
 		{
-			if ($segmentIndex != -1 && ($curCaption["startTime"] < $segmentStartTime || $curCaption["startTime"] >= $segmentEndTime) &&
-				($curCaption["endTime"] < $segmentStartTime || $curCaption["endTime"] >= $segmentEndTime)
+			if ($segmentStartTime > -1 && $segmentEndTime > -1
```

Comment:
Since this code apears twice on this file, I would move it to private function validateVtt(curCaption, segmentStartTime, segmentEndTime)